framework_tvb/tvb/core/services/backend_client.py

```diff
@@ -287,7 +294,295 @@ def stop_operation(operation_id):
+            sleep(10)
+        LOGGER.info(job.properties[HPCSettings.JOB_STATUS_KEY])
+        return job.working_dir
+
+    @staticmethod
+    def _stage_out_to_operation_folder(working_dir, operation):
+        # type: (Storage, Operation) -> list
+        output_list = HPCSchedulerClient.listdir(working_dir, HPCSimulatorAdapter.OUTPUT_FOLDER)
+        operation_dir = HPCSchedulerClient.file_handler.get_project_folder(operation.project, str(operation.id))
+
+        h5_filenames = HPCSchedulerClient._stage_out_outputs(operation_dir, output_list)
+        LOGGER.info(working_dir.properties)
+        LOGGER.info(working_dir.listdir())
+        return h5_filenames
+
+    @staticmethod
+    def _run_hpc_job(operation_identifier):
+        # type: (int) -> None
+        # operation = dao.get_operation_by_id(operation_identifier)
+        # is_group_launch = operation.fk_operation_group is not None
+        # input_gid = json.loads(operation.parameters)['gid']
+        #
+        # job_inputs = HPCSchedulerClient._prepare_input(operation, input_gid)
+        # input_hpc = os.path.join(os.getcwd(), 'input_hpc')
+        # os.makedirs(input_hpc)
+        # for input in job_inputs:
+        #     shutil.copy(input, input_hpc)
+        #
+        # do_operation_launch(input_gid, 1000, is_group_launch)
+
+        operation = dao.get_operation_by_id(operation_identifier)
+        is_group_launch = operation.fk_operation_group is not None
+        simulator_gid = json.loads(operation.parameters)['gid']
+        job = HPCSchedulerClient._launch_job_with_pyunicore(operation, simulator_gid, is_group_launch)
+
+        wd = HPCSchedulerClient._monitor_job(job)
+        h5_filenames = HPCSchedulerClient._stage_out_to_operation_folder(wd, operation)
+        message = HPCSchedulerClient()._update_db_with_results(operation, h5_filenames)
+        LOGGER.debug(message)
+
+    @staticmethod
+    def _stage_out_outputs(h5_path, output_list):
+        # type: (str, dict) -> list
+        result_filenames = []
+        for output_filename, output_filepath in output_list.items():
+            if type(output_filepath) is not unicore_client.PathFile:
+                continue
+            filename = os.path.join(h5_path, os.path.basename(output_filename))
+            output_filepath.download(filename)
+            result_filenames.append(filename)
+        return result_filenames
+
+    @staticmethod
+    def execute(operation_id, user_name_label, adapter_instance):
+        # type: (int, None, None) -> None
+        """Call the correct system command to submit a job to HPC."""
+        # HPCSchedulerClient._run_hpc_job(operation_id)
+
+        thread = threading.Thread(target=HPCSchedulerClient._run_hpc_job,
+                                  kwargs={'operation_identifier': operation_id})
+        thread.start()
+
+    @staticmethod
+    def stop_operation(operation_id):
+        # TODO: implement this use-case
+        pass
+
+
+if TvbProfile.current.hpc.IS_HPC_RUN:
```

Comment:
This file has got much content. Maybe split ?
And here have an explicit  factory method